salt/utils/templates.py

```diff
@@ -59,9 +59,9 @@ def render_tmpl(tmplsrc,
         if 'salt' in kws:
             if 'cmd.run' in kws['salt'] and 'cmd.shell' in kws['salt']:
                 kws['salt']['cmd.run'] = kws['salt']['cmd.shell']
-            if 'run' in kws['salt'].get('cmd', {}) \
-                    and 'shell' in kws['salt'].get('cmd', {}):
-                kws['salt']['cmd']['run'] = kws['salt']['cmd']['shell']
+            if ('cmd' in kws['salt'] and hasattr(kws['salt']['cmd'], 'run')
```

Comment:
[See here](https://github.com/saltstack/salt/blob/ee9e618/salt/renderers/jinja.py#L300) for an explanation of the hasattr/setattr stuffs.